src/toil/cwl/cwltoil.py

```diff
@@ -1757,8 +2406,29 @@ def filtered_secondary_files(unfiltered_secondary_files: dict) -> list:
             final_secondary_files.append(sf)
     return final_secondary_files
 
+def scan_for_unsupported_requirements(tool: Process) -> None:
+    """
+    Scan the given CWL tool for any unsupported optional features.
+
+    If it has them, raise an informative UnsupportedRequirement.
+
+    """
+
+    UNSUPPORTED_REQUIREMENTS = ["InplaceUpdateRequirement"]
```

Comment:
What happened? `toil-cwl-runner` used to support this!

Reply:
This definitely worked for single node execution and via batch schedulers when using a shared filesystem. In other scenarios (non shared filesystems for the job store) I agree that supporting `InplaceUpdateRequirement` doesn't make sense.

Reply:
I wanted to turn this off because I'm reworking Toil to use logic that works for multiple nodes, which ends up actually keeping Directories' contents in Toil instead of on the filesystem. Right now in the implementation I have, the Directory contents are basically packed into strings that Toil passes around, meaning they can't be updated from elsewhere.

I could optimize it to check if the `single_machine` batch system is being used, or any batch system with a file job store, and then find a place for CWL to keep all its stuff in a local temp directory and support InplaceUpdateRequirement, but that seems awkward. I don't *think* the FileJobStore technically presents an interface to get a shared temp directory right now; I'm not really sure I understand how we were ensuring that files or Directories identified by paths weren't deleted when jobs ended before.

What I am hoping we might be able to do eventually is keep Directory contents in mutable files, which could be replaced from other jobs, allowing for side-effect updates from any machine to be visible on any other, with all the race conditions that you would expect in that scenario being exposed to the workflow. But I'm not sure that that semantically matches what `InplaceUpdateRequirement` is supposed to mean; Lon said it was added because people had big files that were too huge to copy, and allowing the side effects without actually being able to prevent the copy is no help to anyone.